TigranKirakosov/hw4.js

```diff
@@ -0,0 +1,194 @@
+'use strict';
+// Class Order
+function Order(customer) {
+  this.customer = customer || 'Officiant :)';
+  this.positions = [];
+}
+
+// Order methods
+Order.prototype.addToOrder = function() {
+  var args = Array.prototype.slice.call(arguments);
+  for (let arg of args) {
+    this.positions.push(arg);
+  }
+};
+Order.prototype.deleteFromOrder = function(position) {
+  return (this.positions = this.positions.filter(function(posToKeep) {
+    return posToKeep !== position;
+  }));
+};
+Order.prototype.showPositions = function() {
+  return this.positions
+    .map(function(position) {
+      return position.foodType;
+    })
+    .join(', ');
+};
+Order.prototype.calcTotalPrice = function() {
+  return `Price is ${this.positions.reduce(function(sum, position) {
+    return sum + position.showPrice();
+  }, 0)} tugrics`;
+};
+Order.prototype.calcTotalCalories = function() {
+  return `Provides ${this.positions.reduce(function(sum, position) {
+    return sum + position.showCalories();
+  }, 0)} calories`;
+};
+Order.prototype.payTheOrder = function() {
+  return Object.freeze(this), Object.freeze(this.positions);
+};
+
+// Class Food
+function Food(foodType) {
+  this.foodType = foodType;

```

Comment:
I see in general all types of food has the same functions - `showCalories`, `showPrice`, `chooseType` and etc. And it implementations are quite similar - try to extract this functions to `Food` class and overwrite it for specific cases (for instance for `Hamburger` - price is sum of 2 parts of `Hamburger`)

Reply:
At first it was not obvious how to manage that but now I discovered a menu feature which represented by second argument for choosing options for these functions